lib/chef/resource/dsc_resource.rb

```diff
@@ -20,16 +20,34 @@
 class Chef
   class Resource
     class DscResource < Chef::Resource
-
       provides :dsc_resource, os: "windows"
 
+      # This class will check if the object responds to
+      # to_text. If it does, it will call that as opposed
+      # to inspect. This is useful for properties that hold
+      # objects such as PsCredential, where we do not want
+      # to dump the actual ivars
+      class ToTextHash < Hash

```

Comment:
We should probably come up with a better long-term solution for this but for now :+1: